pkg/os/os-mapper-provider.go

```diff
@@ -36,25 +45,104 @@ func NewOSMapProvider(client client.Client) *OSMaps {
 
 // GetOSMaps retrieve the OS mapping config map
 func (o *OSMaps) GetOSMaps() (map[string]string, map[string]string, error) {
+	guestOsToCommon := initGuestOsToCommon()
+	osInfoToCommon := initOsInfoToCommon()
+	err := o.updateOsMapsByUserMaps(guestOsToCommon, osInfoToCommon)
+	return guestOsToCommon, osInfoToCommon, err
+}
+
+func (o *OSMaps) updateOsMapsByUserMaps(guestOsToCommon map[string]string, osInfoToCommon map[string]string) error {
 	osConfigMap := &corev1.ConfigMap{}
-	kubevirtNamespace, _ := clientutil.GetNamespace()
+	osConfigMapName := os.Getenv(OsConfigMapName)
+	osConfigMapNamespace := os.Getenv(OsConfigMapNamespace)
+	if osConfigMapName == "" && osConfigMapNamespace == "" {
+		return nil
+	}
+
 	err := o.Client.Get(
 		context.TODO(),
-		types.NamespacedName{Name: osConfigMapName, Namespace: kubevirtNamespace},
+		types.NamespacedName{Name: osConfigMapName, Namespace: osConfigMapNamespace},
 		osConfigMap,
 	)
 	if err != nil {
-		return nil, nil, err
+		return fmt.Errorf(
+			"Failed to read user OS config-map [%s/%s] due to: [%v]",
+			osConfigMapNamespace,
+			osConfigMapName,
+			err,
+		)
 	}
-	guestOsToCommon := make(map[string]string)
-	err = yaml.Unmarshal([]byte(osConfigMap.Data[guestOsToCommonKey]), &guestOsToCommon)
+
+	err = yaml.Unmarshal([]byte(osConfigMap.Data[GuestOsToCommonKey]), &guestOsToCommon)
```

Comment:
Does `yaml.Unmarshall` merge parsed data into the map passed in parameter or replace its content completely?

Reply:
yes, it merged the values. I'll add checks to the test that when a value is overridden, the total sum of elements is greater than the provided map size.